packages/backend/src/peripherals/database/ForcedTradeOfferRepository.ts

```diff
@@ -1,108 +1,135 @@
+    accepted: undefined,
+  }
+  if (
+    row.accepted_at !== null &&
+    row.stark_key_b !== null &&
+    row.position_id_b !== null &&
+    row.submission_expiration_time !== null &&
+    row.nonce !== null &&
+    row.premium_cost !== null &&
+    row.signature !== null
+  ) {
+    return {
+      ...record,
+      accepted: {
+        at: Timestamp(row.accepted_at),
+        nonce: row.nonce,
+        positionIdB: row.position_id_b,
+        premiumCost: row.premium_cost,
+        signature: row.signature,
+        starkKeyB: StarkKey(row.stark_key_b),
+        submissionExpirationTime: row.submission_expiration_time,
+        transactionHash: row.transaction_hash
+          ? Hash256(row.transaction_hash)
+          : undefined,
+      },
+    }
+  } else {
+    return record
+  }
+}
 
-type ForcedTradeOfferRow = ForcedTradeInitialOfferRow &
-  ForcedTradeAcceptedOfferRow
 export class ForcedTradeOfferRepository extends BaseRepository {
   constructor(knex: Knex, logger: Logger) {
     super(knex, logger)
-    this.addInitialOffer = this.wrapAdd(this.addInitialOffer)
-    this.addAcceptedOffer = this.wrapAdd(this.addAcceptedOffer)
-    this.findOfferById = this.wrapFind(this.findOfferById)
-    this.getAllInitialOffers = this.wrapGet(this.getAllInitialOffers)
-    this.getAllAcceptedOffers = this.wrapGet(this.getAllAcceptedOffers)
+    this.add = this.wrapAdd(this.add)
+    this.findById = this.wrapFind(this.findById)
     this.getLatest = this.wrapGet(this.getLatest)
     this.deleteAll = this.wrapDelete(this.deleteAll)
+    this.save = this.wrapSave(this.save)
   }
 
-  async addInitialOffer(
-    offer: Omit<ForcedTradeInitialOfferRecord, 'id'>
-  ): Promise<number> {
-    const row = initialOfferToRow(offer)
+  async add(record: RecordCandidate): Promise<Record['id']> {
+    const row = toRowCandidate(record)
     const [id] = await this.knex('forced_trade_offers')
       .insert(row)
       .returning('id')
-    this.logger.debug({ method: 'add' })
     return id
   }
 
-  async addAcceptedOffer({
-    initialOfferId,
-    acceptedOffer,
-  }: {
-    initialOfferId: number
-    acceptedOffer: ForcedTradeAcceptedOfferRecord
-  }): Promise<number> {
-    const row = acceptedOfferToRow(acceptedOffer)
-    await this.knex('forced_trade_offers')
+  async save(record: Record): Promise<boolean> {
+    const row = toRow(record)
+    const updates = await this.knex('forced_trade_offers')
       .update(row)
-      .where('id', initialOfferId)
-    return initialOfferId
-  }
-
-  async getAllInitialOffers(): Promise<ForcedTradeInitialOfferRecord[]> {
-    const rows = await this.knex('forced_trade_offers').select('*')
-    return rows.map(initialOfferToRecord)
-  }
-
-  async getAllAcceptedOffers() {
-    const rows = await this.knex('forced_trade_offers')
-      .whereNotNull('accepted_at')
-      .select('*')
-
-    return rows.map(acceptedOfferToRecord)
+      .where('id', '=', row.id)
+    return !!updates
   }
 
   async getLatest({
```

Comment:
Don't we need this to be the latest but not the accepted offers?

Reply:
This makes more sense, good catch